Userland/Utilities/image2bin.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/DeprecatedString.h>
+#include <AK/Random.h>
+#include <AK/StringBuilder.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/DirIterator.h>
+#include <LibCore/System.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/Desktop.h>
+#include <LibMain/Main.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath unix"));
+    DeprecatedString path;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(path, "Path to image", "path");
+    args_parser.parse(arguments);
+
+    auto bitmap = TRY(Gfx::Bitmap::try_load_from_file(path));
+
+    TRY(Core::System::pledge("stdio"));
+    Vector<u8> data;
+    for (auto height = 0; height < bitmap->size().height(); height++) {
+        auto* scanline = bitmap->scanline_u8(height);
+        for (auto width = 0; width < bitmap->size().width(); width++) {
```

Comment:
You took an `u8*` and are now iterating over the row using `bitmap->size().width()`, which is in pixels. Each pixel may be 1 through 4 bytes large, depending on the format.

Maybe just copy over the entire row using `bitmap->pitch()`?

Reply:
You can even verify the final size using `bitmap->size_in_bytes()`. Also not sure what the expected output of `image2bin` is: do you want the binary representation of `Gfx::Bitmap`'s datafor whatever `LibGfx` gives you back, or do you want the binary representation of a specific format of `Gfx::Bitmap` (i.e. convert to a specific format first)?

Reply:
I want the binary representation of `Gfx::Bitmap`'s data for whatever `LibGfx` gives me back for however it decodes the image - so technically an equal images encoded (in terms of raw order of pixels and the color of them) in different formats should give me an identical binary file from this utility, with their pixels' data in the right place and when being `cmp`ed, they will be a perfect match.

Reply:
I fixed the bug now, I think. Feel free to indicate otherwise :)